### ISSUE TITLE
ingress always Initializing

### ISSUE THREAD

**Issue by ouyangzhongmin** [rancher/rancher#33005]
can not create ingress , just always stay Initializing

sudo docker run --privileged -d --restart=unless-stopped -p 80:80 -p 8444:443 rancher/rancher:v2.5-head
deployment, service is running successful
---
| Useful | Info |
| :-- | :-- |
|Versions|Rancher `v2.5-14c8c62a05f364adf85523d0afd54b24d9dc4cda-head` UI: `v2.5-dev` |
|Route|`undefined`|

![image](https://user-images.githubusercontent.com/7018141/120586502-3af86e00-c466-11eb-9fda-f664b9080d5d.png)

**Comment by ouyangzhongmin** [comment]
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    field.cattle.io/creatorId: user-d47vs
    field.cattle.io/ingressState: '{"cGlhbm8taW5ncmVzcy9kZWZhdWx0L3hpcC5pby8vcGlhbm8vMTAxMDc=":"deployment:default:piano-api"}'
    nginx.ingress.kubernetes.io/rewrite-target: /.
  creationTimestamp: "2021-06-03T03:54:39Z"
  generation: 3
  labels:
    cattle.io/creator: norman
  managedFields:
  - apiVersion: extensions/v1beta1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .: {}
          f:field.cattle.io/creatorId: {}
          f:field.cattle.io/ingressState: {}
          f:nginx.ingress.kubernetes.io/rewrite-target: {}
        f:labels:
          .: {}
          f:cattle.io/creator: {}
      f:spec:
        f:rules: {}
    manager: Go-http-client
    operation: Update
    time: "2021-06-03T04:00:39Z"
  name: piano-ingress
  namespace: default
  resourceVersion: "13182"
  selfLink: /apis/extensions/v1beta1/namespaces/default/ingresses/piano-ingress
  uid: 8392769e-c977-4b26-8099-f7418d962c1f
spec:
  rules:
  - host: xip.io
    http:
      paths:
      - backend:
          serviceName: ingress-390088f4ce971be9f2d6eb7ef7225b2a
          servicePort: 10107
        path: /piano
        pathType: ImplementationSpecific
status:
  loadBalancer: {}

**Comment by bengray2** [comment]
Same issue, L7 Ingress stuck on Initializing.  
![image](https://user-images.githubusercontent.com/44791530/120914063-ea3f7a00-c6de-11eb-9f83-953773d98590.png)

Version 2.5.8

Started using: 
docker run -d --restart=unless-stopped
-p 80:80 -p 8443:443
--privileged
rancher/rancher:latest

**Comment by bengray2** [comment]
OK, I figured it out.  The rancher 2.5 doco has some issues that cause this on a single node install.

**First Issue:** 
xip.io is dead.  Don't use it anymore.  The doco for 2.5 still refer to it in the examples, but I think it went offline permanantly.  You can change rancher to use nip.io  (see 'xip.io down' - https://github.com/rancher/rancher/issues/32500). 

**Second Issue:** 
I think you have used a single node install using the doco at https://rancher.com/docs/rancher/v2.5/en/installation/other-installation-methods/single-node-docker/.  This will result in the ingress controller being upset that port 80 is in use on the host/node.   You can confirm this by going to Global > <Your Cluster> > System > nginx-ingress-controller > 	nginx-ingress-controller-wgw7k > View Logs and looking for "port 80 is already in use. Please check the flag --http-port"

Clean rancher away using https://rancher.com/docs/rancher/v2.5/en/cluster-admin/cleaning-cluster-nodes/.   At an absolute minimum use: 
```
docker rm -f $(docker ps -qa)
docker rmi -f $(docker images -q)
docker volume rm $(docker volume ls -q)
```

then

` sudo shutdown -r now`

then

```
rm -rf /etc/ceph \
       /etc/cni \
       /etc/kubernetes \
       /opt/cni \
       /opt/rke \
       /run/secrets/kubernetes.io \
       /run/calico \
       /run/flannel \
       /var/lib/calico \
       /var/lib/etcd \
       /var/lib/cni \
       /var/lib/kubelet \
       /var/lib/rancher/rke/log \
       /var/log/containers \
       /var/log/kube-audit \
       /var/log/pods \
       /var/run/calico
```

**(if you don't remove the above dirs, when you rebuild you will have ssl certificate errors)** 

Then use the following command to run rancher, using ports 8080 and 8443 so that the ingress controller can grab 80 and 443 on the single node.

`docker run -d --restart=unless-stopped -p 8080:80 -p 8443:443 --privileged rancher/rancher:latest `


**Now do the following:** 

1. create your cluster - https://rancher.com/docs/rancher/v2.5/en/installation/install-rancher-on-k8s/.  Make sure to give the node the 'etcd', 'Control Plane' and 'Worker' roles.
2. deploy workload - https://rancher.com/docs/rancher/v2.5/en/quick-start-guide/workload/quickstart-deploy-workload-ingress/
3. it should work now

Does anybody know how to get the rancher team to fix the doco for 2.5 to reflect the xip.io and port clash on single node install issue?
